tests/integration/kubernetes/tests_common.sh

```diff
@@ -252,3 +256,47 @@ set_namespace_to_policy_settings() {
 		"${settings_dir}/new-genpolicy-settings.json"
 	mv "${settings_dir}/new-genpolicy-settings.json" "${settings_dir}/genpolicy-settings.json"
 }
+
+policy_tests_enabled() {
+	# The Guest images for these platforms have been built using AGENT_POLICY=yes -
+	# see kata-deploy-binaries.sh.
+	[ "${KATA_HYPERVISOR}" == "qemu-sev" ] || [ "${KATA_HYPERVISOR}" == "qemu-snp" ] || \
+		[ "${KATA_HYPERVISOR}" == "qemu-tdx" ] || [ "${KATA_HOST_OS}" == "cbl-mariner" ]
+}
+
+add_allow_all_policy_to_yaml() {
+	policy_tests_enabled || return 0
+
+	local yaml_file="$1"
+	local resource_kind="$(yq read ${yaml_file} kind)"
+
+	case "${resource_kind}" in
+
+	Pod)
+		info "Adding allow all policy to ${resource_kind} from ${yaml_file}"
+		ALLOW_ALL_POLICY="${ALLOW_ALL_POLICY}" yq write -i "${yaml_file}" \
+			'metadata.annotations."io.katacontainers.config.agent.policy"' \
+			"${ALLOW_ALL_POLICY}"
```

Comment:
@danmihai1 , I recommend to call `set_metadata_annotation` to apply annotation. https://github.com/kata-containers/kata-containers/blob/c163d9f1145b07123cdd77b967cd01784fef0c3f/tests/integration/kubernetes/lib.sh#L177-L193

Reply:
@ChengyuZhu6 thanks! I think that's a good idea but I think we should perform this additional cleanup in a separate PR. I explained my thoughts in https://github.com/kata-containers/kata-containers/issues/9405 .